### PR TITLE
Pr/reactos fix 64bit issues

### PR DESCRIPTION
**Pull request by tkreuzer**
## Purpose
Fix some 64 bit issues

### CODE REVIEW

File: sdk/include/psdk/commdlg.h [hunk 1]
```diff
@@ -495,14 +495,14 @@ typedef struct tagPDW {
...
+	LPARAM lCustData;
 	LPPRINTHOOKPROC lpfnPrintHook;
 	LPSETUPHOOKPROC lpfnSetupHook;
 	LPCWSTR lpPrintTemplateName;
 	LPCWSTR lpSetupTemplateName;
-	HANDLE hPrintTemplate;
-	HANDLE hSetupTemplate;
-} PRINTDLGW,*LPPRINTDLGW;
+    HGLOBAL hPrintTemplate;
+    HGLOBAL hSetupTemplate;
```
Comment (HBelusca) [inline sdk/include/psdk/commdlg.h]: Indentation change.

**Comment by HBelusca** [comment]
Curiosity question: What's the difference between SLIST_ENTRY and SINGLE_LIST_ENTRY?

File: dll/win32/windowscodecs/typeof.h [hunk 2]
```diff
@@ -38,7 +38,11 @@ typedef void (__cdecl typeof(jpeg_set_defaults))(struct jpeg_compress_struct *);
 typedef unsigned int (__cdecl typeof(jpeg_write_scanlines))(struct jpeg_compress_struct *, char **, unsigned int);
 
 typedef void (*png_error_ptr_1)(struct png_struct_def *, const char *);
+#ifdef _WIN64
+typedef void (*png_rw_ptr_1)(struct png_struct_def *, unsigned char *, size_t);
```
Comment (ThFabba) [inline dll/win32/windowscodecs/typeof.h]: I think this used to be broken but we should now be able to use `size_t` for x86 too.

File: dll/win32/windowscodecs/typeof.h [hunk 3]
```diff
@@ -80,7 +84,11 @@ typedef void (__cdecl typeof(png_set_PLTE))(struct png_struct_def *, struct png_
 typedef void (__cdecl typeof(png_set_tRNS))(struct png_struct_def *, struct png_info_def *, const unsigned char *, int, const struct png_color_16_struct *);
 typedef void (__cdecl typeof(png_set_filter))(struct png_struct_def *, int, int);
 typedef void *thandle_t_1;
+#ifdef _WIN64
+typedef size_t (*TIFFReadWriteProc_1)(thandle_t_1, void *, __typeof_intptr);
```
Comment (ThFabba) [inline dll/win32/windowscodecs/typeof.h]: This should return a signed value. The definition resolves to `signed __int64` for x64

File: sdk/include/psdk/commdlg.h [hunk 4]
```diff
@@ -495,14 +495,14 @@ typedef struct tagPDW {
...
-	DWORD lCustData;
+	LPARAM lCustData;
 	LPPRINTHOOKPROC lpfnPrintHook;
 	LPSETUPHOOKPROC lpfnSetupHook;
 	LPCWSTR lpPrintTemplateName;
 	LPCWSTR lpSetupTemplateName;
-	HANDLE hPrintTemplate;
-	HANDLE hSetupTemplate;
-} PRINTDLGW,*LPPRINTDLGW;
+    HGLOBAL hPrintTemplate;
```
Comment (ThFabba) [inline sdk/include/psdk/commdlg.h]: Indentation seems changed.

File: sdk/tools/mkisofs/CMakeLists.txt [hunk 5]
```diff
@@ -84,6 +84,11 @@ if(MSVC)
     # mkisofs uses K&R-style function definitions to support very old compilers.
     # MSVC complains about the resulting foo() vs. foo(void) mismatches.
     add_target_compile_flags(mkisofs "/wd4113")
+
+    if (${ARCH} MATCHES "amd64")
```
Comment (ThFabba) [inline sdk/tools/mkisofs/CMakeLists.txt]: Prefer STREQUAL

File: drivers/storage/class/cdrom_new/cdrom.c [hunk 6]
```diff
@@ -1412,7 +1412,7 @@ CdRomStartIo(
         //
 
         ExInterlockedPushEntrySList(&(cdData->Mmc.DelayedIrps),
-                                    (PSINGLE_LIST_ENTRY)&(Irp->Tail.Overlay.DriverContext[0]),
+                                    (PSLIST_ENTRY)&(Irp->Tail.Overlay.DriverContext[0]),
```
[the remark this answers is not in the document]
Comment (tkreuzer) [reply drivers/storage/class/cdrom_new/cdrom.c]: Done

File: drivers/storage/class/cdrom_new/mmc.c [hunk 7]
```diff
@@ -1233,7 +1233,7 @@ CdRompFlushDelayedList(
     IN BOOLEAN CalledFromWorkItem
     )
 {
-    PSINGLE_LIST_ENTRY list;
+    PSLIST_ENTRY list;
```
[the remark this answers is not in the document]
Comment (tkreuzer) [reply drivers/storage/class/cdrom_new/mmc.c]: Done

File: sdk/include/psdk/commdlg.h (hunk 1 above)
Comment (tkreuzer) [reply sdk/include/psdk/commdlg.h]: Fixed

File: dll/win32/windowscodecs/typeof.h (hunk 2 above)
Comment (tkreuzer) [reply dll/win32/windowscodecs/typeof.h]: Fixed

File: dll/win32/windowscodecs/typeof.h (hunk 3 above)
Comment (tkreuzer) [reply dll/win32/windowscodecs/typeof.h]: Fixed

File: sdk/tools/mkisofs/CMakeLists.txt (hunk 5 above)
Comment (tkreuzer) [reply sdk/tools/mkisofs/CMakeLists.txt]: Fixed

File: sdk/include/psdk/commdlg.h (hunk 4 above)
Comment (tkreuzer) [reply sdk/include/psdk/commdlg.h]: Fixed

**Pull request merged**